src/clj/puppetlabs/services/jruby/jruby_internal.clj

```diff
@@ -139,39 +146,42 @@
 (schema/defn ^:always-validate
   cleanup-pool-instance!
   "Cleans up and cleanly terminates a JRubyInstance and removes it from the pool."
-  [{:keys [scripting-container pool] :as instance} :- JRubyInstance]
+  [{:keys [scripting-container pool] :as instance} :- jruby-schemas/JRubyInstanceSchema
+   shutdown-fn :- IFn]
   (.unregister pool instance)
-  ;; TODO: need to add support for a callback hook, so that consumers like
-  ;; puppet-server can do their own cleanup.
-  ;(.terminate jruby-puppet)
+  (shutdown-fn instance)
```

Comment:
This would end up being called `cleanup-fn` as well, I presume, if you agree with my suggestion above.